ui/dashboards/src/context/DatasourceStoreProvider.tsx

```diff
@@ -126,18 +131,20 @@ export function DatasourceStoreProvider(props: DatasourceStoreProviderProps) {
     const { results, addResult } = buildListDatasourceMetadataResults(datasourcePluginMetadata.display.name);
 
     // Start with dashboard datasources that have highest precedence
-    if (dashboardResource.spec.datasources !== undefined) {
-      for (const selectorName in dashboardResource.spec.datasources) {
-        const spec = dashboardResource.spec.datasources[selectorName];
-        if (spec === undefined || spec.plugin.kind !== datasourcePluginKind) continue;
-        addResult(spec, selectorName);
+    if (dashboardResource) {
+      if (dashboardResource.spec.datasources !== undefined) {
+        for (const selectorName in dashboardResource.spec.datasources) {
+          const spec = dashboardResource.spec.datasources[selectorName];
+          if (spec === undefined || spec.plugin.kind !== datasourcePluginKind) continue;
+          addResult(spec, selectorName);
+        }
       }
-    }
 
-    // Now look at project-level datasources
-    for (const datasource of datasources) {
-      const selectorName = datasource.metadata.name;
-      addResult(datasource.spec, selectorName);
+      // Now look at project-level datasources
+      for (const datasource of datasources) {
```

Comment:
I don´t understand, why this for loop is under the if (dashboardResource) if it is checking the project level datasources ?